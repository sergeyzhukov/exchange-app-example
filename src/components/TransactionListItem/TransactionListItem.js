import React, { PureComponent } from 'react'
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'

export default class TransactionListItem extends PureComponent {
  render() {
    const { title, amount, onPress } = this.props

    return (
      <TouchableOpacity style={styles.container} onPress={onPress}>
        <View style={styles.nameContainer}>
          <Text style={styles.name}>{title}</Text>
        </View>
        <Text style={styles.amount}>{amount}</Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginHorizontal: 16,
    marginVertical: 16,
    alignItems: 'center',
    flexDirection: 'row',
  },
  valuesContainer: {
    alignItems: 'flex-end',
  },
  nameContainer: {
    flex: 1,
  },
  name: {
    fontSize: 14,
    color: 'black',
    letterSpacing: -0.2,
  },
  amount: {
    fontSize: 18,
    fontWeight: '500',
    color: 'black',
  },

  changeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  change: {
    fontSize: 10,
    marginLeft: 2,
    color: '#777777',
  },
  indicatorUp: {
    tintColor: 'green',
    transform: [
      { scaleY: -1 },
    ],
  },
  indicatorDown: {
    tintColor: 'red',
  },
  last: {
    fontSize: 15,
    marginBottom: 4,
    color: '#333333',
  },
  high: {
    fontSize: 10,
    color: '#777777',
  },
})
