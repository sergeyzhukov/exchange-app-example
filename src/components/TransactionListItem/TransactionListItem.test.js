import React from 'react'
import renderer from 'react-test-renderer'
import TransactionListItem from './TransactionListItem'

const WITHDRAW_TRANSACTION = {
  operation: 'withdrawal',
  amount: 33,
  timestamp: 1,
  symbol: '$',
  account: 'USD',
}

const RECEIVE_TRANSACTION = {
  operation: 'receive',
  amount: 33,
  timestamp: 1,
  symbol: '$',
  account: 'USD',
}

test('render <TransactionListItem /> with receive tx', () => {
  const component = <TransactionListItem transaction={RECEIVE_TRANSACTION} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render <TransactionListItem /> with withdraw tx', () => {
  const component = <TransactionListItem transaction={WITHDRAW_TRANSACTION} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render <TransactionListItem /> with withdraw tx and showAccount', () => {
  const component = <TransactionListItem transaction={WITHDRAW_TRANSACTION} showAccount />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
