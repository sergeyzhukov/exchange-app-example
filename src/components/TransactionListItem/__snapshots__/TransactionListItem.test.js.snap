// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <TransactionListItem /> with receive tx 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderRadius": 20,
        "height": 40,
        "marginRight": 12,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "letterSpacing": -0.2,
        }
      }
    >
      
      Incoming transfer
    </Text>
    <Text
      style={
        Object {
          "color": "#999999",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      04:00:00
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 17,
      }
    }
  >
    +
    33.00
    $
  </Text>
</View>
`;

exports[`render <TransactionListItem /> with withdraw tx 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderRadius": 20,
        "height": 40,
        "marginRight": 12,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "letterSpacing": -0.2,
        }
      }
    >
      
      Outgoing transfer
    </Text>
    <Text
      style={
        Object {
          "color": "#999999",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      04:00:00
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 17,
      }
    }
  >
    -
    33.00
    $
  </Text>
</View>
`;

exports[`render <TransactionListItem /> with withdraw tx and showAccount 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderRadius": 20,
        "height": 40,
        "marginRight": 12,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "letterSpacing": -0.2,
        }
      }
    >
      (USD) 
      Outgoing transfer
    </Text>
    <Text
      style={
        Object {
          "color": "#999999",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      04:00:00
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 17,
      }
    }
  >
    -
    33.00
    $
  </Text>
</View>
`;
